src/entity/active_model.rs

```diff
@@ -22,6 +24,7 @@ where
     ActiveValue::set(v)
 }
 
+/// Defines an unset operation on an [ActiveValue]
```

Comment:
Could some documentation be added covering why this takes an unused `Option<bool>`? I don't feel the type's purpose is very clear, even having dug through the source (off-topic for this PR, but would this perhaps be better served by either a 3-value enum or an `Option` of a 2-value enum? that would allow this to be more self-documenting)

Reply:
Documenting that is a little bit tricky since the field of `ActiveValue` are private. I think a user would have to build the documentation with `--document-private-items` set to see any inner struct documentation.

I have tried to provide an overview of the internals of `ActiveValue`and a code snippet in the documentation. Take a look at commit 30814f0053bf0cfe6a8a0001cf63d31600926802

Reply:
Hi there, welcome! You have made a good point.

It certainly is trying to mock a enum, where `Set` & `Unset` etc is like `Ok` and `Err`. It could as well be `Unset(())`.

I was waiting for private enum variant https://github.com/rust-lang/rfcs/pull/2028#issuecomment-308241043 to hide the `Unchanged` variant from public use.

And that's why.

Reply:
> I have tried to provide an overview of the internals of `ActiveValue`and a code snippet in the documentation. Take a look at commit [30814f0](https://github.com/SeaQL/sea-orm/commit/30814f0053bf0cfe6a8a0001cf63d31600926802)

Thank you

Reply:
in the case of

```rust
fn Unset<V>(_: Option<bool>)
//         ^^^^^^^^^^^^^^^
```

Why the unused parameter? I understand that Unset is supposed to emulate a variant of an enum but that still doesn't explain to me why the parameter.

Reply:
Because `Unset()` is not a valid enum. I agree this is a little confusing. May be I should change it in `0.5` to something like

```rust
enum ActiveValue {
    Set(Value),
    UnchangedDoNotUse,
    NotSet,    
}
```

Reply:
@tyt2y3 something like that would be a positive change, yes. Tbh if the argument has no meaning, the name `Unset` seems fine to me (albeit slightly ambiguous if you mean "unset" (verb, action, to switch to a no longer set state)  or "unset" (adjective, state, not currently set)), I mostly think just getting rid of the argument is enough for it to make sense to me. Thanks!